src/main/java/bishakh/psync/SyncService.java

```diff
@@ -17,69 +18,82 @@
     private static String mapFileServerDirectory = "/home/alarm/DMS/";
     private static String databaseAndLogDirectory = "/home/alarm/DMS/";
     private static String databaseName = "fileDB.txt";
+    private static String contactName ="contactHistoryFile.txt";
+
 
     public Logger logger;
     public WebServer webServer;
     public Discoverer discoverer;
     public FileManager fileManager;
     public FileTransporter fileTransporter;
     public Controller controller;
+   
+
+    public SyncService() throws IOException {
+
 
-    public SyncService() {
         logger = new Logger(databaseAndLogDirectory, PEER_ID);
-        discoverer = new Discoverer(BROADCAST_IP, PEER_ID, PORT, logger);
+        discoverer = new Discoverer(BROADCAST_IP, PEER_ID, PORT, logger,contactName,databaseAndLogDirectory);
         fileManager = new FileManager(PEER_ID, databaseName, databaseAndLogDirectory, syncDirectory, mapFileServerDirectory, logger);
         fileTransporter = new FileTransporter(syncDirectory, logger);
         controller = new Controller(discoverer, fileManager, fileTransporter, syncInterval, maxRunningDownloads, logger, 2, true);
         webServer = new WebServer(8080, controller, logger);
+
     }
 
 
-    public SyncService(String inputPeerId, String baseDirectory) {
-        syncDirectory = baseDirectory + File.separator + "sync" + File.separator;
+    public SyncService(String inputPeerId, String baseDirectory) throws IOException {
+        syncDirectory = baseDirectory  + "sync" + "/";
         mapFileServerDirectory = baseDirectory;
         databaseAndLogDirectory = baseDirectory;
         PEER_ID = inputPeerId;
 
+
         logger = new Logger(databaseAndLogDirectory, PEER_ID);
-        discoverer = new Discoverer(BROADCAST_IP, PEER_ID, PORT, logger);
+        discoverer = new Discoverer(BROADCAST_IP, PEER_ID, PORT, logger,contactName,databaseAndLogDirectory);
         fileManager = new FileManager(PEER_ID, databaseName, databaseAndLogDirectory, syncDirectory, mapFileServerDirectory, logger);
         fileTransporter = new FileTransporter(syncDirectory, logger);
         controller = new Controller(discoverer, fileManager, fileTransporter, syncInterval, maxRunningDownloads, logger, 2, true);
         webServer = new WebServer(8080, controller, logger);
     }
 
-    public SyncService(String inputPeerId, String baseDirectory, int priorityMethod) {
-        syncDirectory = baseDirectory + File.separator + "sync" + File.separator;
+    public SyncService(String inputPeerId, String baseDirectory, int priorityMethod) throws IOException {
+        syncDirectory = baseDirectory +"sync" + "/";
```

Comment:
? why this?

Reply:
Suppose my baseDirectory is provided as "C:/Users/Desktop/WorkingDirectory/" (with a trailing '/').
So, no need to give File.separator after baseDirectory unless syncDirectory would be like "C:/Users/Desktop/WorkingDirectory/ \ sync\".
And I just mentioned '/' instead of File.separator, it does not create problem.

Reply:
Ok,
I just found this https://stackoverflow.com/questions/2417485/difference-between-file-separator-and-slash-in-paths

Reply:
ok,
I just found this
https://stackoverflow.com/questions/2417485/difference-between-file-separator-and-slash-in-paths

Reply:
Okay then I am replacing last '/' with File.separator